lib/publiccloud/instance.pm

```diff
@@ -408,7 +408,7 @@ sub wait_for_ssh {
         last if (not isok($exit_code) and $args{wait_stop});    # ssh port closed ok
 
         # skip SLES4SAP as incompatible with get_public_ip
-        unless (get_var('PUBLIC_CLOUD_SLES4SAP')) {
+        if (!get_var('PUBLIC_CLOUD_SLES4SAP')) {
```

Comment:
Let's keep the `unless` unless you want to remove it otherwise.  Otherwise you can remove the unless. :)